.github/workflows/aws-deploy.yaml

```diff
@@ -0,0 +1,76 @@
+name: aws-deploy
+
+on:
+  pull_request:
+    branches: '*'
+  push:
+    branches: [main, develop]
+
+jobs:
+
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: pre-commit/action@v2.0.0
+
+  sceptre-deploy:
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Install pipenv
+        run: |
+          pip install pipenv
+      - name: Cache Pipenv virtualenv
+        uses: actions/cache@v2
+        id: pipenv-cache
+        with:
+          path: ~/.local/share/virtualenvs
+          key: ${{ runner.os }}-pipenv-v2-${{ hashFiles('**/Pipfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pipenv-v2-
+      - name: Install Dependencies
+        if: steps.pipenv-cache.outputs.cache-hit != 'true'
+        env:
+          PIPENV_NOSPIN: 'true'
+          WORKON_HOME: ~/.local/share/virtualenvs
+          PIPENV_CACHE_DIR: ~/.local/share/pipcache
+        run: pipenv install --dev
+      - name: Assume AWS Role (Dev Account)
+        uses: aws-actions/configure-aws-credentials@v1
```

Comment:
I was expecting to see the conditional `if: github.ref == 'refs/heads/develop'` on this step and the following.

Reply:
I can add the `if` condition as long as we can guarantee that no commits will ever find themselves on `main` without first being on `develop`. Otherwise, you might end up with stacks that haven't been deployed to the dev account. 

Omitting the `if` condition seemed safer because on pushes to `main`, sceptre should skip deployment if a stack in `dev` already exists in the dev account, but if it doesn't, then it would "fill in the gaps". 

Not sure if my safer approach is needed though if we prevent commits to main (or are just diligent with not committing `dev` stacks directly to `main`).

Reply:
@BrunoGrandePhD You're right -- there is an opportunity for user error with the model that was suggested to you. It is the same model that is used for Service Catalog. You can see more information about how that development process works [here](https://sagebionetworks.jira.com/wiki/spaces/IT/pages/1122336799/Developer+Bootstrap). The assumption is that one will create templates in config/dev for testing (with merges to develop) and once those are tested, then the user duplicates the templates in prod and merges to main. For reference, also see the [scipool-infra travis build](https://github.com/Sage-Bionetworks/scipool-infra/blob/develop/.travis.yml).

That method is necessary when testing something that has a UI, as Service Catalog does. It is really not necessary if you're only deploying templates. If you're doing that, then a model that follows the pattern of [scicomp-provisioner](https://github.com/Sage-Bionetworks/scicomp-provisioner/tree/master/config) is probably more sensible -- the one difference from scicomp-provisioner being that one would have one set of configs as it does, but would merge to develop branch to test, and merge to main to deploy to the PHI account.

If you want to create your own development model, I think that's ok, but please document the process in a CONTRIBUTING.md file -- we'll need one of those anyway.